### PR TITLE
extern crate -> rust 2018

### PR DESCRIPTION
**Pull request by riendegris**
I need a tool that targets bragi's web API.
This tool needs to be aware of the parameters used by bragi.
These parameters were not publicly exposed.

There is also a function to generate query parameters, so that they can be easily incorporated in, eg reqwest:

```
let bragi_params = Params { ... };
let query_params = autocomplete_params_to_query_params(bragi_params);
Client::new().post(...).query(query_params).send()...
```

An alternative to the function might be to use Into<Vec<String, String>> ?

An alternative would be to use OpenAPI, to open the bragi API to other languages...

### CODE REVIEW

File: libs/bragi/src/routes/autocomplete.rs [hunk 1]
```diff
@@ -125,6 +125,72 @@ impl JsonParams {
     }
 }
 
+/// This function can be used to translate the bragi parameters
+/// into a vector of pairs directly usable in the query part of the URL
+pub fn autocomplete_params_to_query_params(params: Params) -> Vec<(String, String)> {
```
Comment (kinnou02) [inline libs/bragi/src/routes/autocomplete.rs]: I don't really like this function, it's not really the responsibility of bragi to do this, I'am pretty sure we will forget to update it when we add a parameter. 
I don't see anyway to automate this

Comment (riendegris) [reply libs/bragi/src/routes/autocomplete.rs]: :+1:
yeah... I don't like this function either. I will revert opening private modules, and completely separate my little CLI tool from the bragi crate.

**Comment by riendegris** [comment]
I've renamed the pull request to reflect the fact that it is no longer about publishing modules that were previously private, eg route/autocomplete.
What remains in the change, and is still valid, are changes related to moving code to rust 2018 standards.

File: libs/bragi/src/prometheus_middleware.rs [hunk 2]
```diff
@@ -3,9 +3,8 @@
 // because  we want to use the name of the endpoint for retrocompatibility
 // (and as a side effect we also added the 'in flight' queries (but for this we could have used the Registry))
 
-use std::marker::PhantomData;
-use std::sync::Arc;
-use std::time::SystemTime;
+// #[macro_use]
```
Comment (antoine-de) [inline libs/bragi/src/prometheus_middleware.rs]: you can remove those comments

Comment (riendegris) [reply libs/bragi/src/prometheus_middleware.rs]: :+1:

**Pull request merged**